core/src/main/java/io/grpc/DecompressorRegistry.java

```diff
@@ -52,38 +51,70 @@
 @ExperimentalApi("https://github.com/grpc/grpc-java/issues/1704")
 @ThreadSafe
 public final class DecompressorRegistry {
+  public static DecompressorRegistry emptyInstance() {
+    return new DecompressorRegistry();
+  }
 
-  private static final DecompressorRegistry DEFAULT_INSTANCE = new DecompressorRegistry(
-      new DecompressorInfo(new Codec.Gzip(), true),
-      new DecompressorInfo(Codec.Identity.NONE, false));
+  private static final DecompressorRegistry DEFAULT_INSTANCE =
+      emptyInstance()
+      .with(new Codec.Gzip(), true)
+      .with(Codec.Identity.NONE, false);
 
   public static DecompressorRegistry getDefaultInstance() {
```

Comment:
So with this change it's no longer possible to change the default configuration. Configuring JVM-wide would be the most common use case in my mind. So with this change the API seems strange to me. Do we want to mark it `@Internal` for now and re-expose something like it later when we know more of what we want?